time/src/main/java/net/jqwik/time/internal/properties/arbitraries/valueRanges/OfPrecision.java

```diff
@@ -11,19 +11,34 @@
 
 	private final static Set<ChronoUnit> ALLOWED_PRECISIONS = new HashSet<>(Arrays.asList(HOURS, MINUTES, SECONDS, MILLIS, MICROS, NANOS));
 
-	private ChronoUnit precision = DEFAULT;
-	private boolean precisionSet = false;
+	private ChronoUnit precision;
+	private boolean precisionSet;
 
-	public void set(ChronoUnit precision) {
-		setProgrammatically(precision);
-		this.precisionSet = true;
+	public OfPrecision() {
+		this(DEFAULT, false);
 	}
 
-	public void setProgrammatically(ChronoUnit precision) {
+	private OfPrecision(ChronoUnit precision, boolean precisionSet) {
+		this.precision = precision;
+		this.precisionSet = precisionSet;
+	}
+
+	public OfPrecision set(ChronoUnit precision) {
+		return set(precision, true);
+	}
+
+	public OfPrecision setProgrammatically(ChronoUnit precision) {
+		return set(precision, false);
+	}
+
+	private OfPrecision set(ChronoUnit precision, boolean precisionSet) {
 		if (!ALLOWED_PRECISIONS.contains(precision)) {
 			throw new IllegalArgumentException("Precision value must be one of these ChronoUnit values: HOURS, MINUTES, SECONDS, MILLIS, MICROS, NANOS");
 		}
-		this.precision = precision;
+		if (this.precision.equals(precision) && this.precisionSet == precisionSet) {
+			return this;
+		}
+		return new OfPrecision(precision, precisionSet);
 	}
```

Comment:
Is this what you mean?

Reply:
It’s indeed going in the direction I meant, but seeing it it’s unsymmetrical to the Between objects. :-/
So, for symmetry reasons I‘d go back to the previous approach. 

Bringing the logic into OfPrecision, however, should still be an improvement. 

All that forth and back is an implication of asynchronous pull requests. If we were pair-programming, a lot of design ideas could be tried out and would prove themselves worthwhile or unworthy immediately.

Reply:
Isn't it better to change the other classes, too?